### ISSUE TITLE
question: when using autoscaling, should I limit asg to one availability zone?

### ISSUE THREAD

**Issue by shaunc** [terraform-aws-modules/terraform-aws-eks#637]
# I have issues

## I'm submitting a...

* [x] support request - read the [FAQ](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/docs/faq.md) first!

## What is the current behavior?

According to:
https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#common-notes-and-gotchas

launch templates for use with cluster autoscaler should be limited to one availability zone. Is it good to set in the launch template configuration for a particular autoscaling group, (e.g.) `subnets =[ vpc.public_subnets[0]]` (or other subnet) in order to enforce this restriction? Or does the module deal with this in some other way? (Or is the warning obsolete?)

Whatever the answer, it might be good to have a note in your autoscaling docs.

Thank you!

**Comment by max-rocket-internet** [comment]
Sorry but I would say that gotch is wrong.

> Cluster autoscaler does not support Auto Scaling Groups which span multiple Availability Zones

True but only matters if you are using EBS volumes in k8s.

> you will find that AWS unexpectedly terminates nodes without them being drained because of the rebalancing feature.

This is not correct as we disable this process by default: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/local.tf#L58

TLDR; IMO, you don't need to use an ASG per AZ unless you are using EBS volumes for pods 🙂

**Comment by shaunc** [comment]
Thanks! Hmm... still maybe worthwhile putting a note in the docs in case you do try to use EBS volumes (e.g. with a default storage class)?

**Comment by max-rocket-internet** [comment]
> still maybe worthwhile putting a note in the docs

Definitely. PRs welcome 🙂

**Comment by luarx** [comment]
Hi!
I would be interested to know how I should do in that case:
- How can I define an ASG per AZ? I haven't found any way in the docs :(
